src/LineageHelper.ts

```diff
@@ -0,0 +1,193 @@
+import { Edge, EdgeType, isTable, isEdge, Lineage, Table, Column, LineageElement } from "./Lineage";
+
+interface FocusedTable {
+  type: "table";
+  tableId: string;
+}
+
+interface FocusedColumn {
+  type: "column";
+  tableId: string;
+  columnId: string;
+}
+
+interface FocusedEdge {
+  type: "edge";
+  /**
+   * Table/column that is depended on. If `columnId` is undefined then it is a table. Otherwise it is
+   * a column inside the table.
+   */
+  source: {
+    tableId: string;
+    columnId?: string;
+  };
+
+  /**
+   * Dependent table/column. If `columnId` is undefined then it is a table. Otherwise it is
+   * a column inside the table.
+   */
+  target: {
+    tableId: string;
+    columnId?: string;
+  };
+
+  /** Edge type. See #EdgeType */
+  edgeType?: EdgeType;
+}
+
+/**
+ * The element - table, column, edge - that is in focus
+ */
+export type FocusedElement = FocusedTable | FocusedColumn | FocusedEdge;
```

Comment:
Why do we need extra types just for focusing? Can't we return original elements? I.e. for focusing graph we would return another `Lineage<>` array (subgraph of the original graph).

Reply:
@pavelgubin This type is used to provide the LineageHelper with a reference to the element to find / get edges for / get connected elements for. These types contain only those properties that are required to find the element in the graph, no label etc. This type is used only as an input type for the LineageHelper functions. It's a clean union type (using the `type` property) and also includes a way to create a reference to a specific table. This is not possible using the `Table`, `Column` and `Edge` types.

Reply:
Got it, but I'd propose to abstract from focusing use case renaming `Focused<Entity>` to `<Entity>Id` (i.e. `FocusedTable` -> `TableId`).